src/edu/buffalo/cse/jive/finiteStateMachine/parser/expression/temporal/EExpression.java

```diff
@@ -22,7 +30,84 @@ public EExpression(Expression expression) {
 
 	@Override
 	public Boolean evaluate(Context context) {
-		return evaluate(null, context.getCurrentState(), new HashSet<State>(), context.getStates());
+		
+		if(context.isUseMarker())return evaluate(null, context.getCurrentState(), context.getStates());
+		return evaluate2(null, context.getCurrentState(), new HashSet<State>(), context.getStates());
+		

```

Comment:
Using the value of 'useMarker' variable from Context, the wrapper 'evaluate' method determines if it should find the shortest path or just use the old implementation.